test/dependencies/mongo.go

```diff
@@ -0,0 +1,49 @@
+package dependencies
+
+import "github.com/SkynetLabs/skynet-accounts/lib"
+
+var (
+	// DependencyMongoWriteConflictNMessage is the error message with which the
+	// DependencyMongoWriteConflictN will cause a function to fail.
+	DependencyMongoWriteConflictNMessage = "(WriteConflict) This is a WriteConflict caused by DependencyMongoWriteConflictN."
+)
+
+type (
+	// DependencyMongoWriteConflictN causes the caller to fail with a
+	// WriteConflict N times in a row.
+	DependencyMongoWriteConflictN struct {
+		remainingFailures uint
```

Comment:
If we are testing concurrency issues, should this dependency have a mutex? Might be good to prevent future data races. 

Fine as a follow up.

Reply:
Good idea. Added.